### PR TITLE
feat(chocolatey): begin chocolatey build

### PR DESCRIPTION
**Pull request by zlesnr**
relates to: https://github.com/newrelic/newrelic-cli/issues/170

### CODE REVIEW

File: newrelic-cli/tools/VERIFICATION.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿
+Note: Include this file if including binaries you have the right to distribute. 
+Otherwise delete. this file. If you are the software author, you can change this
+mention you are the author of the software.
+
+===DELETE ABOVE THIS LINE AND THIS LINE===
```
Comment (jthurman42) [inline newrelic-cli/tools/VERIFICATION.txt]: ^^

File: newrelic-cli/tools/LICENSE.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿
+Note: Include this file if including binaries you have the right to distribute. 
+Otherwise delete. this file.
+
+===DELETE ABOVE THIS LINE AND THIS LINE===
```
Comment (jthurman42) [inline newrelic-cli/tools/LICENSE.txt]: Can we dynamically pull in the root LISENSE.md instead?

File: build/chocolatey.mk [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	; cd ..
+
+
+chocolatey-build:
+	@echo "=== $(PROJECT_NAME) === [ choco-build     ]: building chocolatey package"
+	@if [ -z "${CHOCOLATEY_API_KEY}" ]; then \
+		echo "Failure: CHOCOLATEY_API_KEY not set" ; \
+		exit 1 ; \
+	fi
+	@curl -sL -o newrelic-cli/NewRelicCLIInstaller.msi https://github.com/newrelic/newrelic-cli/releases/download/v${PROJECT_VER_TAGGED}/NewRelicCLIInstaller.msi
```
Comment (jthurman42) [inline build/chocolatey.mk]: work in `/tmp` instead?  You also added the MSI into git

File: build/chocolatey.mk [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+		exit 1 ; \
+	fi
+	@cd newrelic-cli && \
+		$(CHOCO) push --source https://chocolatey.org/ -k ${CHOCOLATEY_API_KEY} newrelic-cli.${PROJECT_VER_TAGGED}.nupkg \
+	; cd ..
+
+
+chocolatey-build:
+	@echo "=== $(PROJECT_NAME) === [ choco-build     ]: building chocolatey package"
+	@if [ -z "${CHOCOLATEY_API_KEY}" ]; then \
```
Comment (jthurman42) [inline build/chocolatey.mk:20]: Do we need an API key to build the package, or just publish it?

File: Makefile [hunk 5]
```diff
@@ -2,9 +2,9 @@
 # Global vars
 #############################
 PROJECT_NAME := $(shell basename $(shell pwd))
-PROJECT_VER  ?= $(shell git describe --tags --always --dirty | sed -e '/^v/s/^v\(.*\)$$/\1/g') # Strip leading 'v' if found
+PROJECT_VER  ?= $(shell git describe --tags --always --dirty | sed -e '/^v/s/^v\(.*\)$$/\1/g')# Strip leading 'v' if found
 # Last released version (not dirty)
-PROJECT_VER_TAGGED  := $(shell git describe --tags --always --abbrev=0 | sed -e '/^v/s/^v\(.*\)$$/\1/g') # Strip leading 'v' if found
+PROJECT_VER_TAGGED  := $(shell git describe --tags --always --abbrev=0 | sed -e '/^v/s/^v\(.*\)$$/\1/g')# Strip leading 'v' if found
```
Comment (jthurman42) [inline Makefile:7]: The comments are also fine to delete

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -53,6 +53,12 @@ jobs:
         GH_USER_EMAIL: developer-toolkit-team@newrelic.com
         GH_USER_NAME: 'New Relic Developer Toolkit Bot'
 
+    - name: Upload chocolatey package
+      shell: bash
+      run: make chocolatey-publish
```
Comment (jthurman42) [inline .github/workflows/release.yml:58]: Could this not be part of `make release-publish` ?

File: newrelic-cli/newrelic-cli.nuspec [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    <!-- == SOFTWARE SPECIFIC SECTION == -->
+    <!-- This section is about the software itself -->
+    <title>newrelic-cli</title>
+    <authors>New Relic Developer Toolkit Team</authors>
+    <!-- projectUrl is required for the community feed -->
+    <projectUrl>https://github.com/newrelic/newrelic-cli</projectUrl>
+    <!--<iconUrl>http://cdn.rawgit.com/__REPLACE_YOUR_REPO__/master/icons/newrelic-cli.png</iconUrl>-->
+    <!-- <copyright>Year Software Vendor</copyright> -->
+    <!-- If there is a license Url available, it is required for the community feed -->
+    <!-- <licenseUrl>Software License Location __REMOVE_OR_FILL_OUT__</licenseUrl>
```
Comment (jthurman42) [inline newrelic-cli/newrelic-cli.nuspec]: Should add this to our repo LICENSE file

File: newrelic-cli/newrelic-cli.nuspec [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    <requireLicenseAcceptance>true</requireLicenseAcceptance>-->
+    <!--<projectSourceUrl>Software Source Location - is the software FOSS somewhere? Link to it with this</projectSourceUrl>-->
+    <!--<docsUrl>At what url are the software docs located?</docsUrl>-->
```
Comment (jthurman42) [inline newrelic-cli/newrelic-cli.nuspec]: developer.newrelic.com

File: build/package/chocolatey/newrelic-cli.nuspec [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    <!-- If the software is cross-platform, attempt to use the same id as the debian/rpm package(s) if possible. -->
+    <id>newrelic-cli</id>
+    <!-- version should MATCH as closely as possible with the underlying software -->
+    <!-- Is the version a prerelease of a version? https://docs.nuget.org/create/versioning#creating-prerelease-packages -->
+    <!-- Note that unstable versions like 0.0.1 can be considered a released version, but it's possible that one can release a 0.0.1-beta before you release a 0.0.1 version. If the version number is final, that is considered a released version and not a prerelease. -->
+    <version>0.7.0</version>
+    <!-- <packageSourceUrl>Where is this Chocolatey package located (think GitHub)? packageSourceUrl is highly recommended for the community feed</packageSourceUrl>-->
+    <!-- owners is a poor name for maintainers of the package. It sticks around by this name for compatibility reasons. It basically means you. -->
+    <!--<owners>__REPLACE_YOUR_NAME__</owners>-->
```
Comment (jthurman42) [inline build/package/chocolatey/newrelic-cli.nuspec:32]: New Relic

File: build/package/chocolatey/newrelic-cli.nuspec [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+    <!-- == SOFTWARE SPECIFIC SECTION == -->
+    <!-- This section is about the software itself -->
+    <title>newrelic-cli</title>
+    <authors>New Relic Developer Toolkit Team</authors>
+    <!-- projectUrl is required for the community feed -->
+    <projectUrl>https://github.com/newrelic/newrelic-cli</projectUrl>
+    <!--<iconUrl>http://cdn.rawgit.com/__REPLACE_YOUR_REPO__/master/icons/newrelic-cli.png</iconUrl>-->
+    <!-- <copyright>Year Software Vendor</copyright> -->
+    <!-- If there is a license Url available, it is required for the community feed -->
+    <!-- <licenseUrl>Software License Location __REMOVE_OR_FILL_OUT__</licenseUrl>
```
Comment (jthurman42) [inline build/package/chocolatey/newrelic-cli.nuspec:44]: Link to repo LICENSE file

File: build/package/chocolatey/newrelic-cli.nuspec [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+    <requireLicenseAcceptance>true</requireLicenseAcceptance>-->
+    <!--<projectSourceUrl>Software Source Location - is the software FOSS somewhere? Link to it with this</projectSourceUrl>-->
+    <!--<docsUrl>At what url are the software docs located?</docsUrl>-->
```
Comment (jthurman42) [inline build/package/chocolatey/newrelic-cli.nuspec:47]: developer.newrelic.com

File: build/package/chocolatey/newrelic-cli.nuspec [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+    <!--<mailingListUrl></mailingListUrl>-->
+    <!--<bugTrackerUrl></bugTrackerUrl>-->
```
Comment (jthurman42) [inline build/package/chocolatey/newrelic-cli.nuspec:49]: GH issues page

File: build/package/chocolatey/tools/LICENSE.txt [hunk 13]
```diff
@@ -0,0 +1,11 @@
+﻿
+Note: Include this file if including binaries you have the right to distribute. 
+Otherwise delete. this file.
+
+===DELETE ABOVE THIS LINE AND THIS LINE===
```
Comment (jthurman42) [inline build/package/chocolatey/tools/LICENSE.txt:5]: Can we edit the nuspec `<files>` section to pull this from the root and delete this file?

File: build/package/chocolatey/tools/VERIFICATION.txt [hunk 14]
```diff
@@ -0,0 +1,13 @@
+﻿
+Note: Include this file if including binaries you have the right to distribute. 
+Otherwise delete. this file. If you are the software author, you can change this
+mention you are the author of the software.
+
+===DELETE ABOVE THIS LINE AND THIS LINE===
```
Comment (jthurman42) [inline build/package/chocolatey/tools/VERIFICATION.txt:6]: ^^

File: build/chocolatey.mk (hunk 4 above)
Comment (zlesnr) [reply build/chocolatey.mk:20]: Just build, thanks.

File: build/chocolatey.mk (hunk 3 above)
Comment (zlesnr) [reply build/chocolatey.mk]: MSI deleted.  Can work in /tmp, but simpler not to.

File: newrelic-cli/tools/LICENSE.txt (hunk 2 above)
Comment (zlesnr) [reply newrelic-cli/tools/LICENSE.txt]: Can just copy it in during the build.

File: build/package/chocolatey/newrelic-cli.nuspec (hunk 12 above)
Comment (zlesnr) [reply build/package/chocolatey/newrelic-cli.nuspec]: Done.

File: build/package/chocolatey/tools/LICENSE.txt (hunk 13 above)
Comment (zlesnr) [reply build/package/chocolatey/tools/LICENSE.txt]: I just copied the file in during build.

File: build/package/chocolatey/tools/VERIFICATION.txt (hunk 14 above)
Comment (zlesnr) [reply build/package/chocolatey/tools/VERIFICATION.txt]: Updated.

File: .github/workflows/release.yml (hunk 6 above)
Comment (zlesnr) [reply .github/workflows/release.yml:58]: Maybe yes.  We require that the windows build exists there first before we try to curl it down to package it, so this seemed like a safe place to ensure the rest of the build had been done.

File: build/package/chocolatey/newrelic-cli.nuspec (hunk 10 above)
Comment (zlesnr) [reply build/package/chocolatey/newrelic-cli.nuspec]: Done.

File: build/package/chocolatey/newrelic-cli.nuspec (hunk 9 above)
Comment (zlesnr) [reply build/package/chocolatey/newrelic-cli.nuspec]: Done.

**Pull request merged**